cmds/core/dmesg/dmesg_test.go

```diff
@@ -5,24 +5,53 @@
 package main
 
 import (
-	"os"
+	"bytes"
+	"fmt"
 	"testing"
 
 	"github.com/u-root/u-root/pkg/testutil"
 )
 
 func TestDmesg(t *testing.T) {
-	if uid := os.Getuid(); uid != 0 {
-		t.Skipf("test requires root on CircleCI, your uid is %d", uid)
+	testutil.SkipIfNotRoot(t)
+	for _, tt := range []struct {
+		name      string
+		clear     bool
+		readClear bool
+		want      error
+	}{
+		{
+			name:      "both flags set",
+			clear:     true,
+			readClear: true,
+			want:      fmt.Errorf("cannot specify both -clear and -read-clear"),
+		},
+		{
+			name:      "clear log",
+			clear:     true,
+			readClear: false,
+			want:      fmt.Errorf(""),
+		},
+		{
+			name:      "clear log after printing",
+			clear:     false,
+			readClear: true,
+			want:      fmt.Errorf(""),
+		},
+	} {
+		t.Run(tt.name, func(t *testing.T) {
+			*clear = tt.clear
+			*readClear = tt.readClear
+			buf := &bytes.Buffer{}
+			if got := dmesg(buf); got != nil {
```

Comment:
Can you add a test that buf.String() has stuff in it in the cases where no flag is set and where *readClear?